tracer/src/Datadog.Trace/Debugger/Instrumentation/AsyncMethodDebuggerInvoker.cs

```diff
@@ -45,23 +45,23 @@ public static class AsyncMethodDebuggerInvoker
         /// <param name="typeHandle">The handle of the type</param>
         /// <param name="methodMetadataIndex">The index used to lookup for the <see cref="MethodMetadataInfo"/> associated with the executing method</param>
         /// <param name="isReEntryToMoveNext">If it the first entry to the state machine MoveNext method</param>
-        /// <returns>Live debugger state</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static AsyncMethodDebuggerState BeginMethod<TTarget>(string probeId, int probeMetadataIndex, TTarget instance, RuntimeMethodHandle methodHandle, RuntimeTypeHandle typeHandle, int methodMetadataIndex, ref AsyncMethodDebuggerState isReEntryToMoveNext)
+        public static void BeginMethod<TTarget>(string probeId, int probeMetadataIndex, TTarget instance, RuntimeMethodHandle methodHandle, RuntimeTypeHandle typeHandle, int methodMetadataIndex, ref AsyncDebuggerState isReEntryToMoveNext)
```

Comment:
I think this is a breaking change with the new signature, for the compatibility reasons I brought up in other comments. For improvements, you'll need to add new methods and invoke those from the native side while keeping this API untouched

Reply:
To illustrate the scenario a little more, let's assume we make this change and release it in v2.28.0.

Scenario:
- User has .NET Core app with assembly reference to this latest v2.28.0 Datadog.Trace
- User has set up automatic instrumentation with an older version (e.g. v2.25.0) of Datadog.Trace 
- If the user executes a method that has a method probe, then the v2.25.0 native side will create a MethodReference with the old signature (`AsyncMethodDebuggerState` return type). When the IL is invoked, it will try to load to invoke that method from the v2.28.0 Datadog.Trace (the app already has an assembly reference) but it cannot find the method because the signature has changed so a `MissingMethodException` occurs

To ensure this scenario will still work, we will need to keep the old API around

Reply:
Ignore my API breakage comment here 😄